chapters/arrays.tex

```diff
@@ -1417,12 +1417,12 @@ \section{Empty Arrays}\label{empty-arrays}
 
 Empty matrices can be constructed with the fill function.  For example:
 \begin{lstlisting}[language=modelica]
-Real A[:,:] = fill(0.0, 0, 1); // a Real 0 x 1 matrix
-Boolean B[:, :, :] = fill(false, 0, 1, 0); // a Boolean 0 x 1 x 0 matrix
+Real A[:,:] = fill(0.0, 0, 1); // an empty Real 0 x 1 matrix
+Boolean B[:, :, :] = fill(false, 0, 1, 0); // an empty Boolean 0 x 1 x 0 matrix
 \end{lstlisting}
 
-It is not possible to access an element of an empty matrix, e.g.\ \lstinline!v[j, k]! cannot be evaluated if \lstinline!v = []! because the assertion fails
-that the index must be bigger than one.
+It is not possible to access the elements of an empty matrix, e.g.\ \lstinline!v[j, k]! cannot be evaluated if \lstinline!v = []! because the assertion fails
+that the indices must be greater than zero.
```

Comment:
It would probably be better to rewrite the sentence completely, as there are no _scalar_ elements in an empty matrix/array.  However, by not specifying all subscripts the array may actually be both empty and indexable at the same time.  Maybe it's more interesting to illustrate this point?
```suggestion
\begin{example}
Whereas indexing into an empty dimension of an array is an error, not all applications of indices to empty arrays are invalid:
\begin{lstlisting}
Real[1, 0] a = fill(0.0, 1, 0); // empty 1 x 0 matrix
Real[0] a1 = a[1]; // empty vector
\end{lstlisting}
\end{example}
```

(Note: The suggestion above was written without testing it.)

Reply:
I think the more common and important case would be `:` and range-indexing; such as `a[1, :]` and `a[1, 1:end]`.

We could use the terminology of 10.5 (Array) Indexing and say that scalar access is not allowed as there are no scalar elements, but slicing generating arrays can be legal.

Reply:
I edited the suggestion above to also include slicing with `:`.

Reply:
I like @henrikt-ma's suggestion and the examples but I also like that the PR version explicitly says that indices must be greater than zero. I find it more clear. It might be  Another thing in @henrikt-ma's suggestion that I am wondering about is that the text is now part of the example and would appear formatted as a nonnormative text. The original text is part of the formal text. So I am wondering if it should remain as formal text. Or is it OK to lessen the strength/importance of that statement?

Reply:
The upper and lower index bounds are given in https://specification.modelica.org/master/arrays.html#array-dimension-lower-and-upper-index-bounds, so we shouldn't repeat this normatively here.